### PR TITLE
start recreation of project on kubebuilder v4

### PR DESCRIPTION
**Pull request by drewwells**
https://book.kubebuilder.io/migration/migration_guide_gov3_to_gov4

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -200,173 +130,71 @@ endif
...
 
 .PHONY: uninstall
 uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
-	$(KUSTOMIZE) build config/crd | kubectl delete --namespace `cat .id` --ignore-not-found=$(ignore-not-found) -f -
+	$(KUSTOMIZE) build config/crd | $(KUBECTL) delete --ignore-not-found=$(ignore-not-found) -f -
 
 .PHONY: deploy
-deploy-kustomize: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
-	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG_PATH}:${TAG}
-	$(KUSTOMIZE) build config/default | kubectl apply --namespace `cat .id` -f -
```
Comment (EvertonCalgarotto) [inline Makefile:212]: how do you specify the namespace to deploy this now that "--namespace was removed"?
We need it so more than 1 person can run integration tests in parallel

File: Makefile [hunk 2]
```diff
@@ -200,173 +130,71 @@ endif
...
+deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
+	$(KUSTOMIZE) build config/default | $(KUBECTL) apply -f -
 
 .PHONY: undeploy
-undeploy-kustomize: ## Undeploy controller from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
-	$(KUSTOMIZE) build config/default | kubectl delete	--namespace `cat .id` --ignore-not-found=$(ignore-not-found) -f -
```
Comment (EvertonCalgarotto) [inline Makefile:216]: same here with namespace

File: Makefile [hunk 3]
```diff
@@ -137,60 +63,64 @@ vet: ## Run go vet against code.
...
+
+.PHONY: lint
+lint: golangci-lint ## Run golangci-lint linter
+	$(GOLANGCI_LINT) run
+
+.PHONY: lint-fix
+lint-fix: golangci-lint ## Run golangci-lint linter and perform fixes
+	$(GOLANGCI_LINT) run --fix
 
-.PHONY: integration-test
```
Comment (EvertonCalgarotto) [inline Makefile:142]: how do we run integration tests in the new code?

File: cmd/main.go [hunk 4]
```diff
@@ -113,66 +126,92 @@ func main() {
...
+		Scheme: scheme,
+		Metrics: metricsserver.Options{
+			BindAddress:   metricsAddr,
+			SecureServing: secureMetrics,
+			TLSOpts:       tlsOpts,
+		},
+		WebhookServer:          webhookServer,
+		HealthProbeBindAddress: probeAddr,
 		LeaderElection:         enableLeaderElection,
-		LeaderElectionID:       "32151587.atlas.infoblox.com",
```
Comment (EvertonCalgarotto) [inline cmd/main.go:145]: Is it ok to change the electionid? what processes creates the leaders in k8s?

Comment (drewwells) [reply cmd/main.go:145]: didn't notice that. Probably fine to change it, but reverting this back

File: Makefile (hunk 3 above)
Comment (drewwells) [reply Makefile:142]: they don't run right now. I need to make a helm install process to setup the chart prior to running the existing tests

File: Makefile (hunk 1 above)
Comment (EvertonCalgarotto) [reply Makefile:212]: dbc_end2end_test.go (line 58) relies on this namespace

File: Makefile (hunk 2 above)
Comment (drewwells) [reply Makefile:216]: kubebuilder v3 vs v4 code. It didn't do the right thing before and won't work after either. We just have it b/c kubebuilder template creates it

**Review by EvertonCalgarotto — APPROVED** [review]

**Pull request merged**